lua/hawtkeys/ts.lua

```diff
@@ -67,31 +67,46 @@ local scannedFiles = {}
 ---@param mapDefs keyMapArgs[]
 ---@return string
 local function build_dot_index_expression_query(mapDefs)
-    local query = "(function_call"
-    query = query .. "\nname: (dot_index_expression) @exp (#any-of? @exp "
-    for name, opts in pairs(mapDefs) do
-        if opts.method == "dot_index_expression" then
-            query = query .. ' "' .. name .. '"'
+    local query = [[
+    (function_call
+    (dot_index_expression) @exp (#any-of? @exp %s)
+        (arguments) @args)
+    ]]
+
+    ---@param params keyMapArgs[]
+    ---@return string
+    local function buildArgs(params)
+        local args = ""
+        for name, opts in pairs(params) do
+            if opts.method == "dot_index_expression" then
+                args = args .. ' "' .. name .. '"'
+            end
         end
+        return args
     end
-    query = query .. ")"
-    query = query .. "\n(arguments) @args)"
-    return query
+    return string.format(query, buildArgs(mapDefs))
```

Comment:
```suggestion
    return string.format(query, buildArgs(mapDefs, "dot_index_expression"))
```
This way we avoid creating closures for each call, and reuse the buildArgs function